test_scripts/RC/ButtonPress/011_generic_error_if_read_only_response_come_from_hmi.lua

```diff
@@ -0,0 +1,58 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/smartdevicelink/sdl_requirements/issues/9
+-- Use case: https://github.com/smartdevicelink/sdl_requirements/blob/master/detailed_docs/button_press_emulation.md
+-- Item: Use Case 1: Main Flow
+--
+-- Requirement summary:
+-- [SDL_RC] Button press event emulation
+--
+-- Description:
+-- In case:
+-- 1) RC app sends ButtonPress request with valid parameters
+-- 2) and SDL gets response (resultCode: READ_ONLY) from HMI
+-- SDL must:
+-- 1) Respond to App with success:false, "GENERIC_ERROR"
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonRC = require('test_scripts/RC/commonRC')
+
+--[[ Local Variables ]]
+local modules = { "CLIMATE", "RADIO" }
+
+--[[ Local Functions ]]
+local function getDataForModule(pModuleType, self)
+  local cid = self.mobileSession1:SendRPC("ButtonPress", {
+    moduleType = pModuleType,
+    buttonName = commonRC.getButtonNameByModule(pModuleType),
+    buttonPressMode = "SHORT"
+  })
+
+  EXPECT_HMICALL("Buttons.ButtonPress", {
+    appID = self.applications["Test Application"],
+    moduleType = pModuleType,
+    buttonName = commonRC.getButtonNameByModule(pModuleType),
+    buttonPressMode = "SHORT"
+  })
+  :Do(function(_, data)
+      self.hmiConnection:SendError(data.id, data.method, "READ_ONLY", "Info message")
+    end)
+
+  self.mobileSession1:ExpectResponse(cid, { success = false, resultCode = "GENERIC_ERROR", info = "Info message" })
```

Comment:
`GENERIC_ERROR` is an odd choice here... doesn't appear to be part of the use case either. https://github.com/smartdevicelink/sdl_requirements/blob/master/detailed_docs/RC/button_press_emulation.md

Reply:
agree

Reply:
Since this is for the ButtonPress RPC, it seems that Core is just considering this to be an invalid response from the HMI. `GENERIC_ERROR` is appropriate here.